gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/LTFDocuments.java

```diff
@@ -0,0 +1,130 @@
+import java.nio.file.Path;
+import java.util.Enumeration;
+import java.util.Iterator;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+import org.apache.commons.io.IOUtils;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LTFDocuments implements Iterable<Article>, Closeable {
+
+  private static final Logger log = LoggerFactory.getLogger(LTFDocuments.class);
+
+  private ZipFile zipFile;
+
+  private Enumeration<? extends ZipEntry> ltfZipEntries;
+
+
+  @Override
+  public void close() throws IOException {
+    zipFile.close();
+  }
+
+  @NotNull
+  @Override
+  public Iterator<Article> iterator() {
+    return new ArticlesIterator();
+  }
+
+  private LTFDocuments(ZipFile zipFile) {
+    this.zipFile = zipFile;
+    this.ltfZipEntries = zipFile.entries();
+
+  }
+
+  static LTFDocuments fromLTFZippedFile(Path p) throws IOException {
+
+    ZipFile zipFile = new ZipFile(p.toFile(), StandardCharsets.UTF_8);
+    return new LTFDocuments(zipFile);
+
+  }
+
+
+  private class ArticlesIterator extends AbstractIterator<Article> {
+    private LtfReader reader = new LtfReader();
+    private ZipEntry currentLtfEntry;
+    private ImmutableList<LtfDocument> entryDocs;
+    private int docIndex;
+
+    private ArticlesIterator() {
+      // get first zip entry of Ltf from zip file
+      currentLtfEntry = nextLtfEntry();
+      try {
+        entryDocs = getDocsForLtfEntry(currentLtfEntry);
+        docIndex = 0;
+      } catch (IOException e) {
+        log.error("Error reading zip file entry: {} in {}", currentLtfEntry.getName(), zipFile.getName());
+        log.error(e.getMessage());
+        System.exit(1);
+      }
+    }
+
+    private ZipEntry nextLtfEntry() {
+      while (ltfZipEntries.hasMoreElements()) {
+        ZipEntry entry = ltfZipEntries.nextElement();
+        if (entry.getName().endsWith(".ltf.xml")) {
+          return entry;
+        }
+      }
+      return null; // no more entries
+    }
+
+    private ImmutableList<LtfDocument> getDocsForLtfEntry(ZipEntry entry) throws IOException {
+      try (InputStream is = zipFile.getInputStream(entry)) {
+        byte[] bytes = IOUtils.toByteArray(is);
+        LctlText lctlText =
+            reader.read(ByteSource.wrap(bytes).asCharSource(StandardCharsets.UTF_8));
+        return lctlText.getDocuments();
+      }
+    }
+
+    @Override
+    protected Article computeNext() {
+      if (docIndex >= entryDocs.size()) { // done with this entry
+        currentLtfEntry = nextLtfEntry();
+        if (currentLtfEntry == null) {
+          return endOfData();
+        } else {
+          try {
+            entryDocs = getDocsForLtfEntry(currentLtfEntry);
+            docIndex = 0;
+          } catch (IOException e) {
+            log.error("Error reading zip file entry: {} in {}", currentLtfEntry.getName(), zipFile.getName());
+            log.error(e.getMessage());
+            System.exit(1);
```

Comment:
Wrap the checked exception in `RuntimeException`, don't terminate the program

Reply:
Changed to `RuntimeException`